ares/md/controller/port.cpp

```diff
@@ -20,6 +20,7 @@ auto ControllerPort::unload() -> void {
 }
 
 auto ControllerPort::allocate(string name) -> Node::Peripheral {
+  device = nullptr;
```

Comment:
This shouldn't be required; the call to disconnect() in the parent Node class should clean this up.

Reply:
At this time, it is required b/c disconnect() isn't hooked in, and the code is not resetting device except on port unload(). This looks to be the case for many other controller ports as they were written the same way. In this case, allocate() is working as a makeshift connect() since configuration is done automatically by the device.